### ISSUE TITLE
Missing items in the Map Editor

### ISSUE THREAD

**Issue by aimdev** [jbelien/MapFile-Generator#7]
Loaded latest via composer.

The OUTPUT FORMAT sub section is not visible in the Editor, but exists in the MAP file.

MAP OUTPUTFORMAT
    DRIVER AGG/PNG8
    EXTENSION "png"
    FORMATOPTION "PALETTE=grey.txt"
    FORMATOPTION "PALETTE_FORCE=on"
    IMAGEMODE RGB
    MIMETYPE "image/png;mode=8bit,grayscale"
    NAME "png_bw"
    TRANSPARENT ON
  END # OUTPUTFORMAT

The WEB sub section is not visible in the Editor, but exists in the MAP file.

WEB
    IMAGEPATH /tmp/ms_tmp/
    METADATA
      "ows_enable_request" "*"
      "wms_enable_request" "*"
      "wms_srs" "EPSG:900913 EPSG:4326 EPSG:3857"
      "labelcache_map_edge_buffer" "-10"
      "wms_title" "Natural Earth World WMS"
    END # METADATA
  END # WEB
   
These items not visible in the Editor, but exist in the MAP file.

  CONFIG "ON_MISSING_DATA" "IGNORE"
  CONFIG "MS_ERRORFILE" "../mserror"
  CONFIG "PROJ_LIB" "../"
  DEBUG 5
  FONTSET "../fonts.lst"
  IMAGECOLOR 20 20 20
  IMAGETYPE png
  MAXSIZE 4000

AimDev

**Comment by jbelien** [comment]
Indeed, there is at the moment, no way to edit the parameters you mentioned.

It will be a bit "difficult" to add absolutely everything in the user interface so I focus on the "*main*" parameters.
What would you consider as "*main*" parameters ?

**Comment by aimdev** [comment]
Hi

Yes understood. 
To actually get a map file to load, as opposed to getting a blank map :), then
your are about 95% there.

The ones I highlighted in the issue are required, after that then its really not too much of an issue
to use an ordinary editor.

Will you be providing a export/write the map to a file?, at the moment I am cut and pasting the map
from the mapfile window into an editor then saving.

Thanks

AimDev

**Comment by aimdev** [comment]
Hi
Produced a map file, which initially would not load, blank map.
Support for  SHAPEPATH is required, when I hand edited the map file it loaded ok.

AimDev

**Comment by jbelien** [comment]
> Will you be providing a export/write the map to a file?, at the moment I am cut and pasting the map from the mapfile window into an editor then saving.

Excellent idea !

> Produced a map file, which initially would not load, blank map.
> Support for SHAPEPATH is required, when I hand edited the map file it loaded ok.

`SHAPEPATH` is only required if you don't use the full path in `LAYER` : `DATA`.
Bu I can add a `SHAPEPATH` field in the user interface.

**Comment by aimdev** [comment]
Hi

Yes its better for the shape file directory to be defined in one place.
My mapfile is a combination of shapefiles and postgresql queries.

AimDev
